### ISSUE TITLE
Current sorting mechanism is not natural

### ISSUE THREAD

**Issue by Eisa01** [jellyfin/jellyfin#2661]
### Describe the bug
Episodes without IndexNumber have their sorting order unnatural, and episodes that contain IndexNumber sort special characters to be first.

***Example: I have a series and the episodes metadata are not parsed,e.g.:  no IndexNumber:***
```
[FooSubs] Special E01
[SomeGroup] Special E02
Sample E01
//..Sample E02 to E08
Sample E09
Sample E10
Sample E100
Sample E101
//..Sample E102 to E108
Sample E109
Sample E11
Sample E110
Sample E111
```

***Example: I have a series and the episodes metadata were parsed,e.g.:  contains IndexNumber:***
```
[FooSpecial 1]
[SomeSpecial 2]
1. Sample Title
2. Another Title
3. Different Title
```
### Expected behavior
Series that contain episodes without metadata should be sorted in alphanumeric order, which also disaplays special character to be after normal characters.

***Lets say I have a series and the episodes metadata are not parsed,e.g.:  no IndexNumber:***
```
Sample E01
Sample E02
//..Sample E03 to E98
Sample E99
Sample E100
//..Sample E101 to E320
[FooSubs] Special E01
[SomeGroup] Special E02
```

***Example: I have a series and the episodes metadata were parsed,e.g.:  contains IndexNumber:***
```
1. Sample Title
2. Another Title
3. Different Title
[FooSpecial 1]
[SomeSpecial 2]
```

Above is proper sorting for both series episodes that were selected to not have metadata and that have metadata parsed, in addition to the special characters being displayed last which is usual to avoid spoilers. 

### Additional context
As I have some little experience in development, I have fixed it in the website (client-side), by using the alphanumeric sort order by DaveKoelle.

**In itemdetails.js where season is identified, I have added the following:**
```
else if ("Season" === item.Type) {
          if (!result.Items[0].hasOwnProperty("IndexNumber")) {//does alphanumerc sort for episodes without IndexNumber
		result.Items.sort((a,b)=>alphanum(a.Name,b.Name));
          }
 html = listView.getListViewHtml({
                        items: result.Items,
                        showIndexNumber: false,
...
```
The results are exactly as described in the expected behavior section.

However, I was sucessful only in the client-side, since I was lost in the server side and could not find exactly how the sort method / class is utilized.

I will attempt to fix the current sorting, if someone could lend me a hand.
The sorting is being called by ItemSortBy.SortName, however I am not sure how to edit the SortName code as I cannot find it.

### Screenshots
tl;dr, well screenshots..:)
- Current sorting that has issues: [https://imgur.com/a/lT1yoL7](https://imgur.com/a/lT1yoL7)
- The fixed sorting which is implemented above: [https://imgur.com/a/Pfg4NRc](https://imgur.com/a/Pfg4NRc)
### System
 - OS: [Any]
 - Clients: [Any]
 - Browser: [Any]
 - Jellyfin Version: [10.5.2]
 - Reverse Proxy: [Apache2]
 - Base URL: [yes: /web/index.html#!/itemdetails.html?id..]

**Comment by JustAMan** [comment]
I believe your server-side journey should start somewhere here:

https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Controller/Entities/BaseItem.cs#L630-L649

**Comment by Eisa01** [comment]
@JustAMan 
I was able to manage fixing the sort with the help of a friend, but the way I am doing it seems like a hack, won't be good to make a pull request for it.

Thank you for helping, perhaps this will come handy when I try to approach it in a natural way.

**Comment by JustAMan** [comment]
You can make a draft PR, maybe someone would be able to help with transforming it into something not-a-hack ;)

**Comment by Eisa01** [comment]
Haha gotcha ^ ^
Will see when I am able to manage to set it up for a pr request
